### ISSUE TITLE
SET_NAMED is going to be deprecated

### ISSUE THREAD

**Issue by jangorecki** [Rdatatable/data.table#3729]
SET_NAMED is going to be deprecated in R C API, we have one use of it:
https://github.com/Rdatatable/data.table/blob/29c270b92cd3f33c8898c2e3cbef5ced98b296ab/src/wrappers.c#L71

**Comment by mattdowle** [comment]
Commenting out this one call to setmutable : 
https://github.com/Rdatatable/data.table/blob/master/R/data.table.R#L2393
results in just two tests failing :  510, 1689.1
To be honest I don't remember why this is done. From the comment :
```
// called from one single place at R level. TODO: avoid somehow, but fails tests without
// At least the SET_NAMED() direct call is passed 0 and makes no assumptions about >0.  Good enough for now as patch for CRAN is needed.
```
It looks like we should just try again.  Maybe that was done before we updated to depend on R 3.1.
I don't remember anything fundamental about needing the SET_NAMED. Looks like I tried to remove it before.  Just need to get those tests passing without it.
